cmd/local-locker.go

```diff
@@ -26,17 +26,23 @@ import (
 
 // lockRequesterInfo stores various info from the client for each lock that is requested.
 type lockRequesterInfo struct {
-	writer          bool      // Bool whether write or read lock.
-	node            string    // Network address of client claiming lock.
-	serviceEndpoint string    // RPC path of client claiming lock.
-	uid             string    // UID to uniquely identify request of client.
-	timestamp       time.Time // Timestamp set at the time of initialization.
-	timeLastCheck   time.Time // Timestamp for last check of validity of lock.
+	Writer          bool      `json:"writer"`          // Bool whether write or read lock.
+	Node            string    `json:"node"`            // Network address of client claiming lock.
+	ServiceEndpoint string    `json:"serviceendpoint"` // RPC path of client claiming lock.
+	UID             string    `json:"uid"`             // UID to uniquely identify request of client.
+	Timestamp       time.Time `json:"timestamp"`       // Timestamp set at the time of initialization.
+	TimeLastCheck   time.Time `json:"timelastcheck"`   // Timestamp for last check of validity of lock.
+	Source          string    `json:"source"`          // Contains line, function and filename reqesting the lock.
+}
+
+// LocksMap stores various info from the client for each lock that is requested.
+type LocksMap struct {
+	LockMap map[string][]lockRequesterInfo
```

Comment:
You can have a simpler type:
type LockMap map[string][]lockRequesterInfo

Reply:
Looks like you have defined this struct to be used for RPC response. In that case you can move this structure to admin-rpc-server.go and rename it to TopLocksResp

Reply:
done